controllers/vrg_volrep.go

```diff
@@ -1899,16 +1890,31 @@ func (v *VRGInstance) checkPVClusterData(pvList []corev1.PersistentVolume) error
 	return nil
 }
 
-func (v *VRGInstance) restoreClusterDataObjects(objList []client.Object, objType string) (int, error) {
+func restoreClusterDataObjects[
+	ObjectType any,
```

Comment:
I have never seen this block of code before with a function.
I wonder what `func f[x,y]()` mean? I probably should just google it, but I can't even think of what to ask.

Reply:
It's a [generic](https://go.dev/doc/tutorial/generics), similar to a C++ template.  A function is written once but supports multiple types.  It resolves types at compile-time which is preferable to runtime type checking.